### ISSUE TITLE
Support lifecycle additions on controller - for hooks

### ISSUE THREAD

**Issue by EFX-PXT1** [haproxytech/helm-charts#40]
Be great if the chart had support for lifecycle to add hooks to the controller.
Personal example being a preStop for graceful shutdown.

```
controller:
  lifecycle:
    # graceful shutdown
    preStop:
      exec:
        command: ["/bin/sh", "-c", "kill -USR1 $(pidof haproxy); while killall -0 haproxy; do sleep 1; done"]
```

But generally useful to support.

**Comment by dkorunic** [comment]
Makes sense, I'll add this to feature queue.

**Comment by EFX-PXT1** [comment]
I have it coded - just assigning a number.
Do I need to replicate in both kubernetes-ingress and haproxy.
Not clear on what is going on - can you advise a little @dkorunic  ?
Thanks

**Comment by dkorunic** [comment]
@EFX-PXT1 These are two separate Charts (kubernetes-ingress vs haproxy), not really related -- kubernetes-ingress Chart is for people running HAProxy CE or EE as Ingress Controller as opposed to haproxy Chart which is for people using HAProxy CE or EE as a regular application with the ability to load-balance, but otherwise not as a global Ingress Controller.

**Comment by EFX-PXT1** [comment]
Thanks for the clarity - appreciated.

**Comment by dkorunic** [comment]
Merged #41 , closing.
